django/core/management/commands/migrate.py

```diff
@@ -147,6 +147,11 @@ def handle(self, *args, **options):
         else:
             targets = executor.loader.graph.leaf_nodes()
 
+        for i, target in enumerate(targets):
+            if target in executor.loader.replacements and target not in executor.loader.graph.nodes:
+                incomplete_migration = executor.loader.replacements[target]
+                targets[i] = incomplete_migration.replaces[-1]
```

Comment:
This should be an issue only when `migration_name` is passed, so I would move it to the `if options['app_label'] and options['migration_name']` branch, e.g.
```diff
diff --git a/django/core/management/commands/migrate.py b/django/core/management/commands/migrate.py
index 9ad1854a2d..aa1a41e78c 100644
--- a/django/core/management/commands/migrate.py
+++ b/django/core/management/commands/migrate.py
@@ -140,18 +140,20 @@ class Command(BaseCommand):
                 except KeyError:
                     raise CommandError("Cannot find a migration matching '%s' from app '%s'." % (
                         migration_name, app_label))
-                targets = [(app_label, migration.name)]
+                target = (app_label, migration.name)
+                if (
+                    target not in executor.loader.graph.nodes and
+                    target in executor.loader.replacements
+                ):
+                    incomplete_migration = executor.loader.replacements[target]
+                    target = incomplete_migration.replaces[-1]
+                targets = [target]
```